core/src/test/java/com/linecorp/armeria/client/ClientOptionsBuilderTest.java

```diff
@@ -147,4 +147,45 @@ public void testShortcutMethods() {
         assertThat(opts.responseTimeoutMillis()).isEqualTo(2000);
         assertThat(opts.maxResponseLength()).isEqualTo(3000);
     }
+
+    @Test
+    public void testDecoratorDowncast() {
+        final FooClient inner = new FooClient();
+        final FooDecorator outer = new FooDecorator(inner);
+
+        assertThat(outer.as(inner.getClass())).isPresent();
+        assertThat(outer.as(outer.getClass())).isPresent();
+
+        if (outer.as(inner.getClass()).isPresent()) {
+            assertThat(outer.as(inner.getClass()).get() == inner);
```

Comment:
nit: How about using `isSameAs()` for asserting? :-)
```java
assertThat(outer.as(inner.getClass()).get()).isSameAs(inner);
```

Reply:
@ikhoon 
When I use, `isSameAs()` without `isPresent()`
IDEA warns me
```
Warning:(156, 47) 'Optional.get()' without 'isPresent()' check
```
is it okay?

which one is preferred?
* Use `isSameAs` and leaving warnings
* Use `isSameAs` and suppress warnings using annotation

Reply:
@jyblue Sorry for interrupting, I didn't know IntelliJ IDEA warns it. 
Now, I think this is good for me. 👍

Reply:
@ikhoon Thanks a lot !